### PR TITLE
fix: Audio notification would not work without action on the dashboard

### LINKED ISSUE #4264: Audio notification would not work without a click/action on dashboard
- If I open the dashboard and have not performed any action on the site. (hit the URL, no further action taken on the tab) The audio alerts are not played if even a new conversation/message is created. 
- Audio alerts are played even if I'm active on a chat.

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

Fixes #4264 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/shared/helpers/AudioNotificationHelper.js [hunk 1]
```diff
@@ -47,6 +47,10 @@ export const shouldPlayAudio = (
   userId,
   isDocHidden
 ) => {
+  if (!isDocHidden) {
```
Comment (iamsivin) [inline app/javascript/shared/helpers/AudioNotificationHelper.js:50]: If the tab is not hidden the audio alert will not be available and the favicon badge will be shown for a new message.

**Review by nithindavid — DISMISSED** [review]

**Comment by nithindavid** [comment]
Please test this on both widget and dashboard as we use the same helper at both sides. 
@iamsivin

**Comment by iamsivin** [comment]
> Please test this on both widget and dashboard as we use the same helper at both sides. @iamsivin

Tested it audio is working in both cases if tab hidden and visible.

File: app/javascript/shared/helpers/AudioNotificationHelper.js [hunk 2]
```diff
@@ -47,6 +47,10 @@ export const shouldPlayAudio = (
   userId,
   isDocHidden
 ) => {
+  if (!isDocHidden && message.message_type === MESSAGE_TYPE.INCOMING) {
```
Comment (nithindavid) [inline app/javascript/shared/helpers/AudioNotificationHelper.js:50]: Use camelCase for message type

**Review by nithindavid — CHANGES_REQUESTED** [review]

File: app/javascript/shared/helpers/AudioNotificationHelper.js (hunk 2 above)
Comment (iamsivin) [reply app/javascript/shared/helpers/AudioNotificationHelper.js]: Fixes by https://github.com/chatwoot/chatwoot/pull/4303/commits/99edefbd37bc0c1f0accfd32c96f261307922183

**Review by muhsin-k — APPROVED** [review]

**Review by nithindavid — APPROVED** [review]

**Pull request merged**